utils/src/main/java/software/amazon/awssdk/utils/async/OutputStreamPublisher.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.utils.async;
+
+import static software.amazon.awssdk.utils.CompletableFutureUtils.joinInterruptibly;
+
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.utils.CancellableOutputStream;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Adapts a {@link Publisher} to an {@link OutputStream}.
+ * <p>
+ * Writes to the stream will block until demand is available in the downstream subscriber.
+ */
+@SdkProtectedApi
+public class OutputStreamPublisher extends CancellableOutputStream implements Publisher<ByteBuffer> {
+    private final SimplePublisher<ByteBuffer> delegate = new SimplePublisher<>();
+    private final CountDownLatch subscribedLatch = new CountDownLatch(0);
```

Comment:
Same here, shouldn't it be `new CountDownLatch(1)`?

Reply:
Guess my testing isn't sufficient. Will fix. Good catch.